### ISSUE TITLE
Lua error with pet.health_pct check comparison (boolean against number)

### ISSUE THREAD

**Issue by enzovalentini** [Hekili/hekili#372]
**Describe the Bug/Issue**
I'm getting a spam of LUA Errors occasionally when I mount up on my BM Hunter; the exact error message is below.

**To Reproduce**
1: Leave Combat
2: Mount up (causing pet to despawn)
3: Flood of LUA errors begin and last until Reload UI

**Expected behavior**
No errors?

Message: [string "-- Beast Mastery:default:1..."]:2: attempt to compare boolean with number
Time: Wed Jun  3 19:23:42 2020
Count: 1915
Stack: [string "-- Beast Mastery:default:1..."]:2: attempt to compare boolean with number
[string "=[C]"]: ?
[string "-- Beast Mastery:default:1
return safebool( pet.health_pct<30 )"]:2: in function `Conditions'
[string "@Interface\AddOns\Hekili\Scripts.lua"]:1200: in function `CheckScript'
[string "@Interface\AddOns\Hekili\Core.lua"]:715: in function `GetPredictionFromAPL'
[string "@Interface\AddOns\Hekili\Core.lua"]:1051: in function `GetNextPrediction'
[string "@Interface\AddOns\Hekili\Core.lua"]:1301: in function `ProcessHooks'
[string "@Interface\AddOns\Hekili\UI.lua"]:825: in function <Interface\AddOns\Hekili\UI.lua:689>

**Hekili / Issue Reporting pastebin:**
https://pastebin.com/UEYKAa1m

**Addon Error Message**

Message: [string "-- Beast Mastery:default:1..."]:2: attempt to compare boolean with number
Time: Wed Jun  3 19:23:42 2020
Count: 1915
Stack: [string "-- Beast Mastery:default:1..."]:2: attempt to compare boolean with number
[string "=[C]"]: ?
[string "-- Beast Mastery:default:1
return safebool( pet.health_pct<30 )"]:2: in function `Conditions'
[string "@Interface\AddOns\Hekili\Scripts.lua"]:1200: in function `CheckScript'
[string "@Interface\AddOns\Hekili\Core.lua"]:715: in function `GetPredictionFromAPL'
[string "@Interface\AddOns\Hekili\Core.lua"]:1051: in function `GetNextPrediction'
[string "@Interface\AddOns\Hekili\Core.lua"]:1301: in function `ProcessHooks'
[string "@Interface\AddOns\Hekili\UI.lua"]:825: in function <Interface\AddOns\Hekili\UI.lua:689>

**Comment by Hekili** [comment]
Can you post the first entry in your Beast Mastery priority's default action list?

I'm curious how you entered the pet.health_pct criteria.

Anyway, I made it so pet.health_pct will return 100 when you don't have a pet, which is dumb, but would keep you from getting a Mend Pet recommendation when you don't have a pet up.  Try -07-beta1 and let me know.

That said, if the ability in your Beast Mastery - default - 1 is Mend Pet, the criteria shouldn't even be tested when you don't have a living pet.

**Comment by enzovalentini** [comment]
You are likely on the money here; first priority on Action List default is:
1. Barbed Shot - pet.health_pct < 30

Conditions on the ability:
pet.health_pct < 30

I have no recollection of ever changing the priority as for BM it is absolutely spot on with my own simulations and rotation weak auras.
I have installed -07- and have killed around 20 mobs trying to reproduce the issue and it seems fixed now, thanks for the quick reply! I don't know if I should close this or not depending on the expected, but I just searched the .zip file and didn't find the configuration above anywhere, so I'm scratching my head regarding how this happened. Anyways, thanks again!
